geckomat/get_enzyme_data/loadDatabases.m

```diff
@@ -0,0 +1,148 @@
+function databases = loadDatabases(modelAdapter,selectDatabase)
+% loadDatabases
+%   Load or downloads (if necessary) the organism-specific KEGG and UniProt
+%   databases that are required to extract protein information.
+%
+% Input:
+%   modelAdapter    a GECKO organism-specific modelAdapter function that
+%                   contains the taxonomic ID (taxonID, for UniProt) and
+%                   organism code (keggID, for KEGG)
+%   selectDatabase  which databases should be loaded, either 'uniprot',
+%                   'kegg' or 'both'
+%
+% Output:
+%   databases       contains .uniprot and .kegg structures, dependent on
+%                   which databases were selected.
+%
+% Usage: databases = loadDatabases(modelAdapter,selectDatabase)
+
+if nargin<2
+    selectDatabase = 'both';
+end
+geckoPath = findGECKOroot();
+param=modelAdapter.getParameters();
+keggID=param.keggID;
+taxonID=num2str(param.taxonID);
+
+weboptions('Timeout',10);
+
+%% Uniprot
+if any(strcmp(selectDatabase,{'uniprot','both'}))
+    filePath = fullfile(geckoPath,'databases','uniprot',[num2str(taxonID) '.tsv']);
```

Comment:
I'm thinking then that this would be replaced by something like

filePath = fullfile(modelAdapter.getUniprotPath(),[num2str(taxonID) '.tsv']);

Reply:
And in the base adapter for example  (not tested):

        function path = getUniprotPath(obj)
            geckoPath = findGECKOroot();
            path = fullfile(geckoPath,'databases','uniprot');
        end

Someone should think through if there should be just one function for all paths like I did now (yes I was lazy :) ) or one function per path, I don't have a strong opinion here.

Reply:
This way, that function could be overridden in a TestModelAdapter